src/creator.ts

```diff
@@ -516,7 +533,10 @@ export class SlashCreator extends (EventEmitter as any as new () => TypedEventEm
   cleanRegisteredComponents() {
     if (this._componentCallbacks.size)
       for (const [key, callback] of this._componentCallbacks) {
-        if (callback.expires < Date.now()) this._componentCallbacks.delete(key);
+        if (callback.expires != null && callback.expires < Date.now()) {
+          if (callback.onExpired != null) callback.onExpired();
```

Comment:
This could use optional chaining... but I'm not exactly sure on the implications.
```suggestion
          callback.onExpired?.();
```

Reply:
Unfortunately, changing it to use optional chaining would break the unit tests.
For more information, please refer to [these logs.](https://github.com/Snazzah/slash-create/runs/3763097018)

Reply:
Having an `onExpired` function would imply that there is a timer for components, when it only cleans during each component interaction. Not too sure how this would help when it's not on time.

Reply:
I can provide an example of when the `onExpired` callback would be helpful:
I currently maintain a bot that allows users to query their Genshin Impact in-game user profile. It generates a dropdown to allow users to select their characters to view more details. However, after 15min, the dropdown still stays in the chat but stops reacting to messages since its callback is not registered anymore. Having an `onExpired` callback would allow the bot to send a "final" message or edit the original bot response to replace the components with a "Please rerun the command to interact with it" message.

TLDR: It would allow fewer "hanging" and unreactive components on the chat after the callback's timer has elapsed.